### DISCUSSION TITLE
What is the perf event(cache-misses) sample frequency

### DISCUSSION THREAD

**Discussion by zct** [async-profiler/async-profiler#1019] [General]
When profiling cache-misses,  command like this:
 -d 300 -i 20ms -e cache-misses
` ./profiler.sh -d 300 -i 20ms -e cache-misses`

the code(perfEvents_linux.cpp) set  perf_event attr sample_period  to 20000000（20ms) , does this mean the cache-miss event sample once every 20000000 times?
![img_v3_02fi_5956e3a4-bc8f-43ce-ada2-5d8b8aa1645h](https://github.com/user-attachments/assets/21f8234d-e6db-414d-88af-33530cef2258)

**Comment by apangin** [comment]
Time interval for cache misses and other PMU events does not make sense; specifying `ms` as units is incorrect and will likely be prohibited in future async-profiler versions.

`20m` is treated is 20 million, that is, `-e cache-misses -i 20m` will sample every 20'000'000th cache miss event.

**Comment by zct** [comment]
Thanks for your reply!
use "-i 20m" is correct for pmu events?

**Reply by apangin** [reply]
Yes, that's fine
